restli-server/src/main/java/com/linkedin/restli/internal/server/RestLiRouter.java

```diff
@@ -338,12 +338,39 @@ private ResourceMethodDescriptor findMethodDescriptor(final ResourceModel resour
   private ResourceMethod mapResourceMethod(final ServerResourceContext context,
                                            final ResourceLevel resourceLevel)
   {
+    boolean contextHasActionName = context.getRequestActionName() != null;
+    boolean contextHasFinderName = context.getRequestFinderName() != null;
+    boolean contextHasBatchFinderName = context.getRequestBatchFinderName() != null;
+
+    // Explicitly setting other flags to false if condition meets:
+    if (context.getRequestMethod().equalsIgnoreCase("GET") && contextHasBatchFinderName )
+    {
+      // If "bq" in query params for "GET" methods, then this param considered as BatchFinder name
+      // and "q" and "action" can be used as query param name
+      contextHasActionName = false;
+      contextHasFinderName = false;
+    }
+    else if (context.getRequestMethod().equalsIgnoreCase("GET") && contextHasFinderName)
```

Comment:
Since FINDER came first and BATCH_FINDER was added later, I prefer picking the FINDER param as the first choice.

Reply:
To be honest I think there is no enough reasoning to choose one against another.

Maybe you are referring to the fact that "FINDER" came first so it is more widely used.

But I am not sure what user are truely thinking when they are actually use "bq" and "q" simultanously and what we should put into the doc.

"bq" has a precedence over "q" is my natural thinking since this is the hierarchy in my mind..

Reply:
I decide to follow your comment.